src/test/ui/associated-type-bounds/inside-adt.rs

```diff
@@ -0,0 +1,33 @@
+// compile-fail
+
+#![feature(untagged_unions)]
+
+struct S1 { f: dyn Iterator<Item: Copy> }
+//~^ associated type bounds are not allowed within structs, enums, or unions
+//~| the value of the associated type `Item` (from the trait `std::iter::Iterator`) must be specified [E0191]
```

Comment:
It's sub-optimal, yeah, but difficult to fix as far as I can tell.

Reply:
This and the non-ideal message for E0277 (`T = impl Bounds` rather than `T: Bounds`) should be solved in a follow-up PR. Unfortunately span information is not available at the time this error is emitted, but @estebank should have some ideas hopefully.